tests/tests/force_settle_fee_tests.cpp

```diff
@@ -0,0 +1,1105 @@
+/*
+ * Copyright (c) 2020 Michel Santos, and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <boost/test/unit_test.hpp>
+
+#include <graphene/chain/hardfork.hpp>
+
+#include <graphene/protocol/market.hpp>
+#include <graphene/chain/market_object.hpp>
+
+#include "../common/database_fixture.hpp"
+
+using namespace graphene::chain;
+using namespace graphene::chain::test;
+
+struct force_settle_database_fixture : database_fixture {
+   force_settle_database_fixture()
+           : database_fixture() {
+   }
+
+   /**
+    * Create a smart asset without a force settlement fee percent
+    * @param name Asset name
+    * @param issuer Issuer ID
+    * @param force_settlement_offset_percent Force-settlement offset percent
+    * @return Asset object
+    */
+   const asset_object &create_smart_asset(
+           const string &name,
+           account_id_type issuer /* = GRAPHENE_WITNESS_ACCOUNT */,
+           uint16_t force_settlement_offset_percent /* = 100 */ /* 1% */
+   ) {
+      try {
+         optional<uint16_t> force_settlement_fee_percent; // Not specified
+         return create_smart_asset(name, issuer, force_settlement_offset_percent, force_settlement_fee_percent);
+      } FC_CAPTURE_AND_RETHROW((name)(issuer)(force_settlement_offset_percent))
+   }
+
+
+   /**
+    * Create a smart asset
+    * @param name Asset name
+    * @param issuer Issuer ID
+    * @param force_settlement_offset_percent Force-settlement offset percent
+    * @param force_settlement_fee_percent Force-settlement fee percent (BSIP87)
+    * @return Asset object
+    */
+   const asset_object &create_smart_asset(
+           const string &name,
+           account_id_type issuer /* = GRAPHENE_WITNESS_ACCOUNT */,
+           uint16_t force_settlement_offset_percent /* 100 = 1% */,
+           optional<uint16_t> force_settlement_fee_percent /* 100 = 1% */
```

Comment:
Ideally, we define a default value for this argument, then the function above with one argument fewer can be removed. Current code is acceptable for test cases though.

BTW I think it's better to add new parameters to the functions in `database_fixture` class, rather than adding new functions here. It's not a priority though, we can refactor in the future.

Reply:
Those are reasonable points.  We'll defer the refactoring for now.